include/aws/cal/der.h

```diff
@@ -0,0 +1,249 @@
+#ifndef AWS_C_CAL_DER_H
+#define AWS_C_CAL_DER_H
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/cal/exports.h>
+
+#include <aws/common/array_list.h>
+#include <aws/common/byte_buf.h>
+
+struct aws_der_encoder {
+    struct aws_allocator *allocator;
+    struct aws_byte_buf storage;
+    struct aws_byte_buf *buffer; /* buffer being written to, might be storage, might be a sequence/set buffer */
+    struct aws_array_list stack;
+};
```

Comment:
advantages of declaring this in header file (ABI, added #includes) vs declaring it in c file and forcing user to new() it?

Reply:
Just that you can make one on the stack. So, :man_shrugging: